btsieve/src/ethereum/mod.rs

```diff
@@ -58,8 +75,71 @@ where
             .await
             .unwrap_or_else(|e| log::warn!("Failed to wait for delay: {:?}", e));
 
+        let mut new_missing_block_futures = Vec::new();
+        for (block_future, blockhash) in missing_block_futures.into_iter() {
+            match block_future.await {
+                Ok(Some(block)) => {
+                    let block: ethereum_support::Block<ethereum_support::Transaction> = block;
+                    for transaction in block.transactions.iter() {
+                        let result = blockchain_connector
+                            .receipt_by_hash(transaction.hash)
+                            .compat()
+                            .await;
+
+                        let receipt = match result {
+                            Ok(Some(receipt)) => receipt,
+                            Ok(None) => {
+                                log::warn!("Could not get transaction receipt");
+                                continue;
+                            }
+                            Err(e) => {
+                                log::warn!(
+                                    "Could not retrieve transaction receipt for {}: {:?}",
+                                    transaction.hash,
+                                    e
+                                );
+                                continue;
+                            }
+                        };
+
+                        if pattern.matches(transaction, &receipt) {
```

Comment:
❓📣 
Just wondering: I think to remember that I was the one implementing it :D but please correct me if I'm wrong: 

When we introduced the bloomfilter query, the idea was that we don't have to fetch every `transaction_receipt` for every `transaction` in every `block`. Right now this is not the case, do you know what happened?

Reply:
Explained offline. Issue #1551 tracks some missing optimisations.